### PR TITLE
socket-util: introduce iflabel_valid 

### PR DESCRIPTION
**Pull request by ssahani**
socket-util: introduce iflabel_valid

### CODE REVIEW

**Comment by keszybz** [comment]
In 7454aec7ff3797a346fc926aea2dc615fdb97990 you wrote:
> IFA_LABEL does not need much of a validation except the length that is IFNAMSIZ as seen from kernel code.

Can you explain in the commit message why you are doing the current change?

**Comment by ssahani** [comment]
Addressing @poettering's comment . https://github.com/systemd/systemd/pull/5146

File: src/basic/socket-util.c [hunk 1]
```diff
@@ -900,6 +900,29 @@ bool ifname_valid(const char *p) {
...
+                return false;
+
+        if (dot_or_dot_dot(p))
+                return false;
+
+        while (*p) {
+                if ((unsigned char) *p >= 127U)
+                        return false;
+
+                if ((unsigned char) *p <= 32U)
```
Comment (poettering) [inline src/basic/socket-util.c]: this prohibits spaces, is this really intended? (i honestly don't know)

Comment (poettering) [reply src/basic/socket-util.c]: also, if this is only applicable to addresses, it should probably be called address_label_valid(), no?

Comment (ssahani) [reply src/basic/socket-util.c]: `ip address` command thinks space is a garbage.
```
sudo  ip address add 192.168.99.37/24 brd + dev wlp3s0 label wlp3s0 11
Error: either "local" is duplicate, or "11" is a garbage.
```
yes renamed to `address_label_valid()`

**Comment by ssahani** [comment]
Mistakenly closed opened #5464

File: src/basic/socket-util.c (hunk 1 above)
Comment (poettering) [reply src/basic/socket-util.c]: well, you'd have to enclose the label with "" if you include a space:

```
sudo  ip address add 192.168.99.37/24 brd + dev wlp3s0 label "wlp3s0 11"
```

And if I run that it does accept it

Comment (ssahani) [reply src/basic/socket-util.c]: ohh din't know that 

should change to `  if ((unsigned char) *p <= 31U)` then as ascii of space is 32.

Comment (poettering) [reply src/basic/socket-util.c]: i'd write that as `if ((uint8_t) *p < (uint8_t) ' ')`…

Comment (ssahani) [reply src/basic/socket-util.c]: ok updated https://github.com/systemd/systemd/pull/5464.

**Pull request closed without merging**